src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java

```diff
@@ -0,0 +1,69 @@
+package net.sf.jabref.gui.worker;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import net.sf.jabref.gui.BasePanel;
+import net.sf.jabref.gui.JabRefFrame;
+import net.sf.jabref.gui.undo.NamedCompound;
+import net.sf.jabref.gui.undo.UndoableFieldChange;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.logic.util.DOI;
+import net.sf.jabref.model.FieldChange;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.FieldName;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class LookupDOIsWorker extends AbstractWorker {
+
+    private final JabRefFrame frame;
+    private String message;
+
+    private static final Log LOGGER = LogFactory.getLog(LookupDOIsWorker.class);
+
+    public LookupDOIsWorker(JabRefFrame frame) {
+        this.frame = Objects.requireNonNull(frame);
+    }
+
+    @Override
+    public void run() {
+        BasePanel basePanel = Objects.requireNonNull(frame.getCurrentBasePanel());
+        List<BibEntry> bibEntries = basePanel.getSelectedEntries();
+        if (!bibEntries.isEmpty()) {
+            String totalCount = Integer.toString(bibEntries.size());
+            NamedCompound namedCompound = new NamedCompound(Localization.lang("Look up DOIs"));
+            int count = 0;
+            int foundCount = 0;
+            for (BibEntry bibEntry : bibEntries) {
```

Comment:
I have no idea how to create a progress bar there.

I found http://stackoverflow.com/a/26814087/873282, but that code looks much more complicated than the current code. Should I change it?